### PR TITLE
Improved testing for simulated backends.

### PR DESCRIPTION
**Pull request by jeromekelleher**
- Adds an extraConfig dictionary to `configure` to allow us to easily
  configure the tests.
- Refactored test_views to get rid of malformed requests, and set up
  full request validation.
- Renamed test_end_to_end to test_simulated_stack and removed the
  wormtable dependency there.

Fixes issues #250 and #188.

### CODE REVIEW

File: tests/unit/test_simulated_stack.py [hunk 1]
```diff
@@ -8,41 +8,55 @@
 from __future__ import unicode_literals
 
 import unittest
+from nose.tools import nottest  # TMP until we fix Calls
```
Comment (dcolligan) [inline tests/unit/test_simulated_stack.py]: TMP = temporary?

File: tests/unit/test_simulated_stack.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
 """
```
Comment (dcolligan) [inline tests/unit/test_simulated_stack.py]: Yay on file rename.  This will close #188

**Comment by dcolligan** [comment]
Great job.  +1

File: tests/unit/test_simulated_stack.py (hunk 1 above)
Comment (jeromekelleher) [inline tests/unit/test_simulated_stack.py]: I've changed this to a TODO.

**Pull request merged**